apps/openmw/mwmechanics/aiactivate.cpp

```diff
@@ -23,33 +23,29 @@ namespace MWMechanics
         return new AiActivate(*this);
     }
 
-    bool AiActivate::execute (const MWWorld::Ptr& actor, CharacterController& characterController, AiState& state, float duration)
+    bool AiActivate::execute(const MWWorld::Ptr& actor, CharacterController& characterController, AiState& state, float duration)
     {
-        ESM::Position pos = actor.getRefData().getPosition(); //position of the actor
         const MWWorld::Ptr target = MWBase::Environment::get().getWorld()->searchPtr(mObjectId, false); //The target to follow
 
         actor.getClass().getCreatureStats(actor).setDrawState(DrawState_Nothing);
 
-        if(target == MWWorld::Ptr() ||
-            !target.getRefData().getCount() || !target.getRefData().isEnabled()  // Really we should be checking whether the target is currently registered
+        if (target == MWWorld::Ptr() ||
+            !target.getRefData().getCount() || !target.getRefData().isEnabled()  // Really we should check whether the target is currently registered
                                                                                 // with the MechanicsManager
-                )
-            return true;   //Target doesn't exist
+            )
+        return true;   //Target doesn't exist
 
-        //Set the target desition from the actor
+        //Set the target destination for the actor
         ESM::Pathgrid::Point dest = target.getRefData().getPosition().pos;
 
-        if(distance(dest, pos.pos[0], pos.pos[1], pos.pos[2]) < 200) { //Stop when you get close
-            actor.getClass().getMovementSettings(actor).mPosition[1] = 0;
+        if (pathTo(actor, dest, duration, 200)) //Go to the destination
+        {
+            // activate when reached

```

Comment:
Just a minor thing, but I recently changed the `200` here to `MWBase::Environment::get().getWorld()->getMaxActivationDistance())` in my PR on telekinesis, but I see you are still using 200. It's not a big deal at the moment because AI is set to always trigger a trap if they activate a trapped object anyway, but this was briefly an issue for me when I was working on the code and AI was activating objects outside of the max activation distance (192) and therefore not being hit by traps. It seems to me we might as well use maxactivationdistance or less here anyway, since I don't see why AI actors should be able to activate objects from farther away than the player can.